code/modules/reagents/reagent_containers/food/drinks/bottle.dm

```diff
@@ -73,6 +73,19 @@
 	qdel(src)
 	return B
 
+/obj/item/weapon/reagent_containers/food/drinks/bottle/verb/smash_bottle()
+	set name = "Smash Bottle"
+	set category = "Object"
+
+	var/list/things_to_smash_on = list()
+	for(var/atom/A in range (1, usr))
+		if(A.density && usr.Adjacent(A) && !istype(A, /mob))
+			things_to_smash_on += A
+
+	var/choice = input("Select what you want to smash the bottle on.") as null|anything in things_to_smash_on

```

Comment:
The user may cancel the input which leaves choice as null, need to  add
```
if(!choice)
    return
```

You also need to recheck if the object is still dense and that the user is adjacent after the input, as both states may have changed while the user was considering his options, thus:
```
if(!(choice.density && usr.Adjacent(choice)))
    // Insert options message to user here, about choice not being available anymore.
    return
```